daemon/volumes.go

```diff
@@ -163,11 +165,29 @@ func (container *Container) parseVolumeMountConfig() (map[string]*Mount, error)
 		if m, exists := mounts[mountToPath]; exists {
 			return nil, fmt.Errorf("Duplicate volume %q: %q already in use, mounted from %q", path, mountToPath, m.volume.Path)
 		}
+
+		runtime.LockOSThread()
```

Comment:
Is there any reason to lock the thread here?
It also needs a `defer runtime.UnlockOSThread()`.